comp/core/tagger/component.go

```diff
@@ -16,17 +16,28 @@ package tagger
 import (
 	"context"
 
+	"github.com/DataDog/datadog-agent/comp/core/tagger/telemetry"
 	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
 	taggertypes "github.com/DataDog/datadog-agent/pkg/tagger/types"
 	"github.com/DataDog/datadog-agent/pkg/tagset"
 )
 
 // team: container-platform
 
+// ReplayTagger interface represent the tagger use for replaying dogstatsd events.
+type ReplayTagger interface {
+	Component
+
+	// LoadState loads the state of the replay tagger from a list of entities.
+	LoadState(state []types.Entity)
+}
```

Comment:
Creating this interface and the new component function `ReplayTagger()` allow to acces the internal state of the tagger when building the replay tagger and avoid access the impl package directly outside the comp folder.

Reply:
Because the replay tagger must be a tagger instance as well. Once we create the tagger replay instance we set as the global tagger instance here: 

https://github.com/DataDog/datadog-agent/blob/e2a70b64b2e55226fb7bdb164c03d27d06bd459e/comp/api/api/apiimpl/grpc.go#L134

https://github.com/DataDog/datadog-agent/blob/e2a70b64b2e55226fb7bdb164c03d27d06bd459e/comp/core/tagger/component.go#L51